harness/determined/cli/experiment.py

```diff
@@ -308,6 +308,17 @@ def local_experiment(args: Namespace) -> None:
 
     with det._local_execution_manager(args.model_def.resolve()):
         trial_class = determined.load.trial_class_from_entrypoint(entrypoint)
+        try:
+            from determined import pytorch
+
+            if issubclass(trial_class, pytorch.PyTorchTrial):
+                determined.experimental.test_one_batch_pytorch(
+                    trial_class=trial_class, config=experiment_config
+                )
+                return
+        except ImportError:
+            pass
+
         determined.experimental.test_one_batch(trial_class=trial_class, config=experiment_config)
```

Comment:
I think this pytorch special-case detection should go inside `test_one_batch()`, not here.

2 reasons:
- `det.experimental.test_one_batch()` is technically a public API, albeit an experimental one.  I don't see any value in breaking it and forcing users using it to use `det.experimental.test_one_batch_pytorch()` instead.
- I don't want us to introduce `det.experimental.test_one_batch_pytorch()` into our public API, since it's a legacy-compatibility function.  Users who see it know about it should choose not to use it as soon as they know about it.